File: stockify_backend_golang/src/feature/user/service/user_service_implementation.go
package service

import (
	"stockify_backend_golang/src/feature/user/model"
	"stockify_backend_golang/src/feature/user/repository"
)

type userService struct {
	repo repository.UserRepository
}

func UserServiceImplementation(repo repository.UserRepository) UserService {
	return &userService{repo: repo}
}

func (s *userService) AddUser(user model.User) {
	s.repo.AddUser(user)
}

func (s *userService) GetAllUsers() []model.User {
	return s.repo.GetAllUsers()
}

func (s *userService) GetUserById(id uint64) model.User {
	return s.repo.GetUserById(id)
}

func (s *userService) UpdateUser(user model.User) {
	s.repo.UpdateUser(user)
}

func (s *userService) DeleteUserById(id uint64) {
	s.repo.DeleteUserById(id)
}

func (s *userService) GetFilteredUsers(params model.UserQueryParams) ([]model.User, error) {
	return s.repo.GetFilteredUsers(params)
}

File: stockify_app_flutter/lib/feature/user/service/user_service_implementation.dart
import 'package:stockify_app_flutter/feature/user/model/user.dart';
import 'package:stockify_app_flutter/feature/user/repository/user_repository.dart';
import 'package:stockify_app_flutter/feature/user/service/user_service.dart';

class UserServiceImplementation extends UserService {
  UserServiceImplementation._privateConstructor();

  static final UserService _instance =
      UserServiceImplementation._privateConstructor();

  static UserService get instance => _instance;

  final UserRepository _userRepository = UserRepository.instance;

  @override
  void addUser(User user) {
    _userRepository.addUser(user);
  }

  @override
  void deleteUser(int id) {
    _userRepository.deleteUser(id);
  }

  @override
  void editUser(User user) {
    _userRepository.editUser(user);
  }

  @override
  List<User> getAllUsers() {
    return _userRepository.getAllUsers();
  }

  @override
  User getUser(int id) {
    return _userRepository.getUser(id);
  }
}
